File: python/programATM.py
# Program ATM Sederhana - Python

while True:
    nama = input("Masukkan Nama Anda: ")
    konfirmasi = input(f"Apakah nama '{nama}' sudah benar? (TRUE/FALSE): ").upper()

    if konfirmasi == "TRUE":
        break
    else:
        print("Silakan masukkan nama kembali!\n")

# Input NIM dan konversi jadi saldo
while True:
    nim = input("Masukkan NIM Anda (angka saja, contoh 2310355): ")
    if nim.isdigit():
        saldo = int(nim)
        break
    else:
        print("NIM harus berupa angka!\n")

# Format saldo
saldo_rp = f"Rp {saldo:,.0f}".replace(",", ".")

print("\n===== DATA ANDA =====")
print(f"Nama  : {nama}")
print(f"Saldo : {saldo_rp}")

# Menu ATM
while True:
    print("\n===== MENU ATM =====")
    print("1. Cek Saldo")
    print("2. Tarik Tunai")
    print("3. Setor Tunai")
    print("4. Transfer")
    print("5. Keluar")

    pilihan = input("Pilih menu (1-5): ")

    match pilihan:
        case "1":
            print(f"Saldo Anda saat ini: {saldo_rp}")

        case "2":
            tarik = int(input("Masukkan jumlah yang ingin ditarik: "))
            if tarik <= saldo:
                saldo -= tarik
                saldo_rp = f"Rp {saldo:,.0f}".replace(",", ".")
                print("Penarikan berhasil!")
                print(f"Saldo tersisa: {saldo_rp}")
            else:
                print("Saldo tidak cukup!")

        case "3":
            setor = int(input("Masukkan jumlah setor: "))
            saldo += setor
            saldo_rp = f"Rp {saldo:,.0f}".replace(",", ".")
            print("Setoran berhasil!")
            print(f"Saldo sekarang: {saldo_rp}")

        case "4":
            tujuan = input("Masukkan nomor rekening tujuan: ")
            jumlah = int(input("Masukkan jumlah transfer: "))
            if jumlah <= saldo:
                saldo -= jumlah
                saldo_rp = f"Rp {saldo:,.0f}".replace(",", ".")
                print(f"Transfer ke {tujuan} berhasil!")
                print(f"Saldo tersisa: {saldo_rp}")
            else:
                print("Saldo tidak cukup!")

        case "5":
            print("Terima kasih telah menggunakan ATM!")
            break

        case _:
            print("Pilihan tidak valid, coba lagi!")

File: PHP/programATM.php
<?php
// === 1. Meminta Nama dan Validasi ===
while (true) {
    echo "Masukkan nama Anda: ";
    $nama = trim(fgets(STDIN));

    echo "Apakah nama Anda benar? (TRUE/FALSE): ";
    $valid = strtoupper(trim(fgets(STDIN)));

    if ($valid == "TRUE") {
        break;
    } else {
        echo "Nama salah, silakan masukkan lagi.\n\n";
    }
}

// === 2. Input NIM dan Konversi Menjadi Saldo ===
echo "Masukkan NIM Anda (angka saja): ";
$nim = trim(fgets(STDIN));

$saldo = (int)$nim;

// Format saldo
$saldo_format = number_format($saldo, 0, ',', '.');

// === 3. Tampilkan Nama dan Saldo Awal ===
echo "\n============================\n";
echo "Nama  : $nama\n";
echo "Saldo : Rp $saldo_format\n";
echo "============================\n\n";

// === 4. Menu ATM (Switch–Case) ===
while (true) {
    echo "===== MENU ATM =====\n";
    echo "1. Cek Saldo\n";
    echo "2. Tarik Tunai\n";
    echo "3. Setor Tunai\n";
    echo "4. Transfer\n";
    echo "5. Keluar\n";
    echo "Pilih menu: ";
    $pilih = trim(fgets(STDIN));

    switch ($pilih) {
        case '1':
            echo "Saldo Anda sekarang: Rp " . number_format($saldo, 0, ',', '.') . "\n\n";
            break;

        case '2':
            echo "Masukkan jumlah tarik tunai: ";
            $tarik = trim(fgets(STDIN));
            if ($tarik > $saldo) {
                echo "Saldo tidak cukup!\n\n";
            } else {
                $saldo -= $tarik;
                echo "Tarik tunai berhasil!\n";
                echo "Sisa saldo: Rp " . number_format($saldo, 0, ',', '.') . "\n\n";
            }
            break;

        case '3':
            echo "Masukkan jumlah setor tunai: ";
            $setor = trim(fgets(STDIN));
            $saldo += $setor;
            echo "Setor tunai berhasil!\n";
            echo "Saldo sekarang: Rp " . number_format($saldo, 0, ',', '.') . "\n\n";
            break;

        case '4':
            echo "Masukkan jumlah transfer: ";
            $tf = trim(fgets(STDIN));
            if ($tf > $saldo) {
                echo "Saldo tidak cukup untuk transfer!\n\n";
            } else {
                echo "Masukkan rekening tujuan: ";
                $rek = trim(fgets(STDIN));
                $saldo -= $tf;
                echo "Transfer ke $rek berhasil!\n";
                echo "Sisa saldo: Rp " . number_format($saldo, 0, ',', '.') . "\n\n";
            }
            break;

        case '5':
            echo "Terima kasih telah menggunakan ATM!\n";
            exit;

        default:
            echo "Menu tidak tersedia!\n\n";
    }
}
?>
